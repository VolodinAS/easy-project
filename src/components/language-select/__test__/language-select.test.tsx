/* eslint-disable check-file/folder-naming-convention */
import React from 'react';
import { describe, it, expect } from '@jest/globals';
import { render, fireEvent, screen } from '@testing-library/react';

import LanguageSelect from '../index';

describe('language-select', () => {
    it('button click', () => {
        render(
            <LanguageSelect />
        );
        expect(document.body).toMatchSnapshot();
        fireEvent.click(screen.getByTitle('language select'), {target: {value: 'en'}});
        expect(document.body).toMatchSnapshot();
        fireEvent.click(screen.getByTitle('language select'), {target: {value: 'ru'}});
        expect(document.body).toMatchSnapshot();
    });
});