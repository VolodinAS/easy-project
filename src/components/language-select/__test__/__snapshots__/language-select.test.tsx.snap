// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language-select button click 1`] = `
<body>
  <div>
    <button
      class="css-1640pth"
      title="language select"
      value="ru"
    >
      <img
        class="css-zlc08d"
        src="test-file-stub"
      />
    </button>
  </div>
</body>
`;

exports[`language-select button click 2`] = `
<body>
  <div>
    <button
      class="css-1640pth"
      title="language select"
      value="en"
    >
      <img
        class="css-zlc08d"
        src="test-file-stub"
      />
    </button>
  </div>
</body>
`;

exports[`language-select button click 3`] = `
<body>
  <div>
    <button
      class="css-1640pth"
      title="language select"
      value="ru"
    >
      <img
        class="css-zlc08d"
        src="test-file-stub"
      />
    </button>
  </div>
</body>
`;
